common/shared/src/main/scala/com/wavesplatform/common/state/ByteStr.scala

```diff
@@ -5,14 +5,25 @@ import com.wavesplatform.common.utils.{Base58, Base64}
 import scala.util.Try
 
 final case class ByteStr(arr: Array[Byte]) {
-  lazy val base58: String = Base58.encode(arr)
-  lazy val base64: String = "base64:" + Base64.encode(arr)
-  lazy val trim: String = base58.toString.take(7) + "..."
-  override lazy val toString: String = base58
+  lazy val base58: String    = Base58.encode(arr)
+  lazy val base64Raw: String = Base64.encode(arr)
+  lazy val base64: String    = "base64:" + base64Raw
+  lazy val trim: String = (if (arr.length < 1024) {
+                             base58.toString.take(7)
+                           } else {
+                             base64Raw
+                           }) + "..."
+  override lazy val toString: String = if (arr.length < 1024) {
+    base58
+  } else {
+    base64
+  }
 
   def isEmpty: Boolean =
     arr.length == 0
 
+  def size: Int = arr.length
```

Comment:
Is it used anywhere?

Reply:
[Decimpiler.scala:68](https://github.com/wavesplatform/Waves/pull/2141/files#diff-d27c15004f08b5dd11f2be22e8984380R68)